src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -214,8 +214,11 @@ internal class ConnectionManager : IConnectionManager {
 
         private bool IsValidLocalConnection(string name, string path) {
             try {
-                var info = new RInterpreterInfo(name, path);
-                return info.VerifyInstallation();
+                Uri uri;
+                if (Uri.TryCreate(path, UriKind.Absolute, out uri) && uri.IsFile) {
```

Comment:
I fixed it, it now restores connection. But there is no indication in the UI that something is going on for ~5-10 seconds especially when REPL is closed.